src/GlobalState.lua

```diff
@@ -158,6 +163,35 @@ function M.logistic_del(unit_number)
   global.mod.logistic[unit_number] = nil
 end
 
+function M.is_vehicle_entity(name)
+  return name == "spidertron"
+end
+
+function M.vehicle_scan_surfaces()
+  for _, surface in pairs(game.surfaces) do
+    local entities = surface.find_entities_filtered { name = "spidertron" }
+    for _, entity in ipairs(entities) do
+      M.vehicle_add_entity(entity)
+    end
+  end
+end
+
+function M.get_vehicle_entity(unit_number)
+  return global.mod.vehicles[unit_number]
+end
+
+-- add a vehicle, assume the caller knows what he is doing
+function M.vehicle_add_entity(entity)
+  if global.mod.vehicles[entity.unit_number] == nil then
+    global.mod.vehicles[entity.unit_number] = entity
+    Queue.push(global.mod.scan_queue, entity.unit_number)
```

Comment:
Yeah it makes sense to update spidertrons from the queue rather than on a timer since there can be a ton of spidertrons in a map at once.

Reply:
I'm not sure if that was sarcastic. =)

If you ever split the queue into a high and low priority queue, then vehicles should go in the low priority queue.
Servicing them once every few seconds should be sufficient.

I mainly used the queue because it was easy.

Reply:
Nope, not sarcastic. Mainly comparing how these updates are done vs player updates which are done every second on a timer.

Agree with the low and high priority queues.